src/bin/polymer-bundler.ts

```diff
@@ -224,59 +228,94 @@ if (options.redirect) {
             return {prefix, path};
           })
           .filter((r: redirection) => r.prefix && r.path);
-  const resolvers: UrlResolver[] =
-      redirections.map((r: redirection) => new RedirectionResolver(r.prefix));
+  const resolvers: UrlResolver[] = redirections.map((r: redirection) => {
+    let resolvedPath = Uri.file(path.resolve(r.path)).toString();
+    // Ensure trailing slash if input path had trailing slash
+    if (r.path.match(/\\$|\/$/)) {
+      resolvedPath = ensureTrailingSlash(resolvedPath);
+    }
+    return new RedirectResolver(
+        packageUrlResolver.resolve(r.prefix as PackageRelativeUrl)!,
+        r.prefix,
+        resolvedPath);
+  });
+
+  // Wrap the getPackageRelativeUrl function for each redirection.  Because we
+  // are wrapping the function, we iterate in reverse order to ensure the
+  // redirect transformations are processed in the same order as they're given.
+  redirections.reverse().forEach((r: redirection) => {
```

Comment:
While I am delighted at this new method on Array I didn't know about, I'm not sure I can really apply it as I'm not actually reducing of redirection details, I'm composing a function.

However, per our discussion in person, this could be better if not generated as an n-sized onion of function calls but rather a single function with an internal iterator over the redirects etc.

For the time being I will leave this as-is and its going to be completely removed once the analyzer itself can provide a reliable PackageRelativeUrl for multiple redirects after I update `UrlResolver#relative()` to support undefined return values for fall-thru support in `MultiUrlResolver` context.